api/api.go

```diff
@@ -13,29 +14,43 @@ import (
 	"github.com/jinzhu/gorm"
 	"github.com/netlify/netlify-commerce/conf"
 	"github.com/netlify/netlify-commerce/mailer"
+	"github.com/pborman/uuid"
 	"github.com/rs/cors"
-	"github.com/zenazn/goji/web/mutil"
 )
 
 var bearerRegexp = regexp.MustCompile(`^(?:B|b)earer (\S+$)`)
 
 // API is the main REST API
 type API struct {
 	handler    http.Handler
+	listener   net.Listener
+	port       int
```

Comment:
if you want to use 0 as a port and let the system decide, but still get that back in code you do need this. Otherwise you have to hardcode the port in the test